lib/kernel/src/inet_db.erl

```diff
@@ -1746,16 +1738,17 @@ do_refresh_cache(CacheDb) ->
 do_refresh_cache('$end_of_table', _, _, OldestT) ->
     OldestT;
 do_refresh_cache(Key, CacheDb, Now, OldestT) ->
-    Fun = fun(RR, T) when RR#dns_rr.tm + RR#dns_rr.ttl < Now ->
-		  ets:match_delete(CacheDb, RR),
-		  T;
-	     (#dns_rr{cnt = C}, T) when C < T ->
-		  C;
-	     (_, T) ->
-		  T
-	  end,
     Next = ets:next(CacheDb, Key),
-    OldT = lists:foldl(Fun, OldestT, ets:lookup(CacheDb, Key)),
+    OldT =
+        case ets:lookup(CacheDb, Key) of
+	    [{_,RR}] when RR#dns_rr.tm + RR#dns_rr.ttl < Now ->
+		ets:delete(CacheDb, Key),
+		OldestT;
+	     [{_,#dns_rr{cnt = C}}] when C < OldestT ->
+		C;
+	     _ ->
+		OldestT
```

Comment:
These lines are not covered in our daily tests by running all test suites in Kernel.

Reply:
That makes sense. Everything is serialized on the server, there is no chance something may have been deleted meanwhile. Awesome catch, I will remove this and below.

Reply:
Oh, sorry!  I thought it was a test case problem and merged.
Well, there will be future PR:s to do code cleanup in... ;-)

Reply:
I will clean it up on the PR that removes the deletes and makes it concurrent. :)

Reply:
Good news, those are needed for the PR that makes it concurrent. :) I am submitting it soon!